addons/resource/models/resource_mixin.py

```diff
@@ -208,8 +208,11 @@ def _list_work_time_per_day(self, from_datetime, to_datetime, calendar=None, dom
             for record in records:
                 intervals = all_intervals[record.resource_id.id]
                 record_result = defaultdict(float)
-                for start, stop, _meta in intervals:
-                    record_result[start.date()] += (stop - start).total_seconds() / 3600
+                for start, stop, meta in intervals:
+                    if calendar.flexible_hours:
+                        record_result[start.date()] = meta.duration_hours
+                    else:
+                        record_result[start.date()] += (stop - start).total_seconds() / 3600
```

Comment:
```suggestion
                for start, stop, meta in intervals:
                    record_result[start.date()] += meta.duration_hours
```

Reply:
mh indeed if the attendance length is modified by a leave, it won't be taken into account, so indeed it's best like this. Thanks!